ytfzf

```diff
@@ -704,6 +708,21 @@ scrape_yt () {
 	unset videos_json yt_json yt_html
 }
 
+parse_special_search () {
+    search="$1"
+    number="$(printf "%s" "$search" | grep -Po "[\-0-9]+")"
+    if [ "$search" = "!!" ]; then
+	number="$search_count"
+    elif [ "$number" -lt 0 ]; then
+	#needs to be + because number is negative
+	number="$((search_count + number + 1))"
+    fi
+    if ! cat "${tmp_searches_prefix}${number}"; then
```

Comment:
to check if file exists you could use the text expression like so
```sh
[ -f $filename ]
```

Reply:
this is like this because if cat doesn't return an error, the contents of the file is printed and is returned to where the function is called, otherwise it returns invalid, but using `[ -f $filename ]` it looks like this

```sh
if [ ! -f "${tmp_searches_prefix}${number}" ]; then
    printf "%s" "$search is an invalid lookback"
    clean_up
    exit 2
else
    cat "${tmp_searches_prefix}${number}"
fi```

it's just more explicit